src/js/visual/configurator-prc.js

```
@@ -310,9 +315,31 @@ ripe.ConfiguratorPrc.prototype.update = async function(state, options = {}) {
 ripe.ConfiguratorPrc.prototype.cancel = async function(options = {}) {
     if (this._buildSignature() === this.signatureLoaded || "") return false;
     if (this._finalize) this._finalize({ canceled: true });
+
+    // flushes the complete set of operations that were waiting
+    // for the end of the pre-loading operation
+    await this.flushPending();
     return true;
 };
 
+/**
+ * Executes pending operations that were not performed so as
+ * to not conflict with the tasks already being executed.
+ */
+ripe.ConfiguratorPrc.prototype.flushPending = async function() {
+    while (this.pendingOperations.length > 0) {
+        const operation = this.pendingOperations.shift();
+        const [name, ...args] = operation;
+        switch (name) {
+            case "changeFrame":
+                await this.changeFrame(...args);
+                break;
+            default:

```

Throw on unknown, to avoid silent errors?